### PR TITLE
CFINSPEC-73: Add cgroup resource

### PR DESCRIPTION
**Pull request by ahasunos**
Signed-off-by: Sonu Saha <sonu.saha@progress.com>
## Description

**ISSUE: CFINSPEC-73 — Add cgroup resource**
`cgroup` resource allows to test different parameters values of the cgroup resource controllers available on linux distributions.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Changes made to existing files
- test/helpers/mock_loader.rb (Mock command used in cgroup)

## New files included for cgroup resource
- docs-chef-io/content/inspec/resources/cgroup.md (Documentation of cgroup)
- lib/inspec/resources/cgroup.rb (Resource library of cgroup)
- test/fixtures/cmd/cgget-n-a (Fixtures of cgget output for the cgroup)
- test/fixtures/cmd/cgget-n-r (Fixtures of cgget output for the cgroup for cpuset subsystem)
- test/fixtures/cmd/cgget-n-r-stat (Fixtures of cgget output for the cgroup for memory subsystem)
- test/unit/resources/cgroup_test.rb (Unit test for the resource cggroup)
- 
## Checklist:

- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Review by clintoncwolfe — APPROVED** [review]
Thanks for your diligence on this!

**Comment by ahasunos** [comment]
> Thanks for your diligence on this!

I'm thankful to you for your patience with me. 😄

File: docs-chef-io/content/inspec/resources/cgroup.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+- `'cupset.cpus'` is a property of this resource and a parameter of the cpuset resource controller
+- `'carrotking'` is the name of cgroup directory
+
+## Properties
+
+- All parameters of the cgroup resource controller are valid properties of this resource. Some of them are: `cupset.cpus`, `memory.limit_in_bytes`, `memory.stat`, `freezer.state`, `cpu.stat`, `cpuacct.usage`, `pids.current`, `blkio.throttle.io_service_bytes`, etc.
+
+## Matchers
+- For a full list of available matchers, please visit our [matchers page](https://docs.chef.io/inspec/matchers/).
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/cgroup.md]: @ahasunos - what does it mean, "Please keep this in mind, when comparing configuration entries that are numbers"?

File: docs-chef-io/content/inspec/resources/cgroup.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+- The matchers applicable for this resource are: `eq`, `cmp`, `match`.
+
+### eq
+`eq` tests for exact equality of two values. It fails if types don’t match. Please keep this in mind, when comparing configuration entries that are numbers. For less restrictive comparisons, please use `cmp`
+
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/cgroup.md:44]: Why dont we provide the origin of this resource's documentation?

File: docs-chef-io/content/inspec/resources/cgroup.md (hunk 1 above)
Comment (ahasunos) [reply docs-chef-io/content/inspec/resources/cgroup.md]: Please refer to this example: https://docs.chef.io/inspec/matchers/#eq

File: docs-chef-io/content/inspec/resources/cgroup.md (hunk 2 above)
Comment (ahasunos) [reply docs-chef-io/content/inspec/resources/cgroup.md:44]: Yes, we have provided the link to the full list of matchers at line 40.

> For a full list of available matchers, please visit our [matchers page](https://docs.chef.io/inspec/matchers/).

Do we need to provide again while explaining the matchers specific to this resource? That seems like redundant to me.
Any thoughts, @clintoncwolfe ?

Comment (Vasu1105) [reply docs-chef-io/content/inspec/resources/cgroup.md:44]: @dkumaras  and @ahasunos we never document the matchers that are common to all resources. We just provide link to the matchers doc page.

Comment (Vasu1105) [reply docs-chef-io/content/inspec/resources/cgroup.md:44]: According to me `eq` matcher should be removed from here.

Comment (ahasunos) [reply docs-chef-io/content/inspec/resources/cgroup.md:44]: > @dkumaras and @ahasunos we never document the matchers that are common to all resources. We just provide link to the matchers doc page.

In my opinion, we should mention(if not explain) the applicable matchers specific to the resource, even if it's part of the global matchers.

Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/cgroup.md:44]: Sure

Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/cgroup.md:44]: @clintoncwolfe , @Vasu1105, @ahasunos : Please confirm the rewritten description of **eq** and **cmp** matchers.

Comment (clintoncwolfe) [reply docs-chef-io/content/inspec/resources/cgroup.md:44]: Looks correct to me, thanks

**Review by dkumaras — APPROVED** [review]

File: docs-chef-io/content/inspec/resources/cgroup.md (hunk 2 above)
Comment (Vasu1105) [reply docs-chef-io/content/inspec/resources/cgroup.md:44]: I still do not agree with the point of adding a matcher description which is common to all the resources here again(https://docs.chef.io/inspec/resources/json - this resource and many of such resources has used that matcher). I am still not sure what is so unique in `eq`, `cmp` matcher here in this resource so we are re-defining that. But of course, it's all docs team call.

**Pull request merged**